steps/24_categorical_support/step.md

```diff
@@ -0,0 +1,143 @@
+|:-----------------:|:-------------------------:|:--------------:|:---------------:|
+| NO | NO | NO | normal case, proceeds as usual without error. |
+| NO | NO | YES| informative error message to user that categorical feature is not supported by the estimator |
+| NO | YES| NO | X |
+| NO | YES| YES| X |
+| YES| NO | NO | normal case, proceeds as usual without error. |
+| YES| NO | YES| use categorical columns as detected by feature_kind and proceed. |
+| YES| YES| NO | informative error message that specified column was not present in data.|
+| YES| YES| YES| use categorical columns as detected by feature_kind along with the user specified columns(union) and proceed. |
+
+- in case where specified column was not found, we can also choose to ignore and continue with warning that column was not found.
+
+#### estimators that can be added:
+- xgboost, lightgbm and catboost are some models that have native support for categorical. But method of passing categorical features varies quite a lot between them (I've commented on this in umbrella issue - https://github.com/sktime/sktime/issues/6109). Making a general interface might be challenging.
+- So it might be better to interface them in separate classes inheriting from reduction forecaster class. (similar to darts lightgbm, see - https://unit8co.github.io/darts/generated_api/darts.models.forecasting.lgbm.html)
+
+#### PROS:
+- No internal encoding, comparatively easier to implement.
+- No major changes to core logic.
+- Follows general ML workflow i.e if estimator does not support, then user must preprocess their data(including encoding) and then use it.
+#### CONS:
+- Less number of estimators support categorical natively.
+
+
+---
+---
+
+### Design 2. Perform one hot encoding if not natively supported. 
+
+- `categorical_feature`  param will be present in all models.
+
+    #### How tag is going to be used:
+    ```
+     if cat support tag is false:
+         encode cat features and continue
+     else:
+         continue
+    ```
+
+| **Estimator supports categorical natively** | **Cat feature specified by user** | **Cat feature present in data** | **Outcome** | 
+|:-----------------:|:-------------------------:|:--------------:|:---------------:|
+| NO | NO | NO | normal case, proceeds as usual without error. |
+| NO | NO | YES| encode features internally as detected by feature_kind and continue.
+| NO | YES| NO | error that specified column was not present in data |
+| NO | YES| YES| use categorical features detected by feature kind and those specified by user(union) and proceed to encode |
+| YES| NO | NO | normal case, proceeds without error.|
+| YES| NO | YES| use categorical features detected by feature_kind and proceed|
+| YES| YES| NO | error that specified column was not present in data|
+| YES| YES| YES| use categorical features detected by feature kind and those specified by user(union) and proceed|
+
+#### Estimators that will be affected:
+- all estimators that take exog input
```

Comment:
Clarification: are we limiting the scope of categorical support to only forecasters and to only `X`?

Reply:
> One major one I can think of immediately is univariate becoming multivariate on one-hot encoding.

I think you making leaps of argumentation here, @Abhay-Lejith - you keep presupposing the "one hot encoding" idea, and silently assuming that is the only way to deal with categorical data.

Reply:
@fkiraly I just brought up a certain case where if the user were to one-hot encode y, it could cause issues. I never said it is the only case. Is anything wrong in this?